conformance/tests/gateway-modify-listeners.go

```diff
@@ -55,7 +55,7 @@ var GatewayModifyListeners = suite.ConformanceTest{
 			require.NoErrorf(t, err, "error getting Gateway: %v", err)
 
 			// verify that the implementation is tracking the most recent resource changes
-			kubernetes.GatewayMustHaveLatestConditions(t, s.TimeoutConfig, original)
+			kubernetes.GatewayMustHaveLatestConditions(t, s.Client, s.TimeoutConfig, gwNN)
```

Comment:
should we move this after the namespace ready check but before fetching the "original" gateway? (similar for the other test in this file)

Reply:
It looks like `original` is only being used for spec fields, so shouldn't matter too much one way or another, but I'm fine changing the order for readability.

Reply:
yeah good point, the original Generation field shouldnt change since the resource shouldnt be updated from when we create it, so I'll defer to you if you think it makes the code more readable to fetch the Gateway after checking the status

Reply:
https://github.com/kubernetes-sigs/gateway-api/pull/2203/commits/991056d97d2d192866d08ee5dfa9e66abda2c047